packages/devextreme/js/__internal/grids/grid_core/context_menu/m_context_menu.ts

```diff
@@ -38,7 +38,9 @@ export class ContextMenuController extends modules.ViewController {
       $element = view && view.element();
 
       if ($element && ($element.is($targetElement) || $element.find($targetElement).length)) {
-        $targetCellElement = $targetElement.closest('.dx-row > td, .dx-row > tr');
+        $targetCellElement = $targetElement.closest('.dx-row > td, .dx-row > tr').length
+          ? $targetElement.closest('.dx-row > td, .dx-row > tr')
+          : $targetElement.find('.dx-group-cell').first();
```

Comment:
I suggest to refactor this part of code a little bit:

```suggestion
              const that = this;
              const $targetElement = $(dxEvent.target);
              let menuItems;
              
              each(VIEW_NAMES, function () {
                const viewName = this;
                const view = that.getView(viewName);
                const $viewElement = view && view.element();
                const isTargetElementInsideView = $element && ($element.is($targetElement) || $element.find($targetElement).length);
                
                if(isTargetElementInsideView) {
                  const $targetCellElement = viewName == "headerPanel"
                    ? $targetElement.find('.dx-group-cell')
                    : $targetElement.closest('.dx-row > td, .dx-row > tr')
                    
                  const $targetRowElement = $targetCellElement.parent();
                
                // ...
```